rules/Privatization/Rector/Property/PrivatizeFinalClassPropertyRector.php

```diff
@@ -113,6 +115,30 @@ private function isPropertyVisibilityGuardedByParent(Property $property, Class_
             if ($parentClassReflection->hasProperty($propertyName)) {
                 return true;
             }
+
+            $classLike = $this->astResolver->resolveClassFromName($parentClassReflection->getName());
+            if (! $classLike instanceof Node\Stmt\ClassLike) {
+                continue;
+            }
+
+            $methods = $classLike->getMethods();
+            foreach ($methods as $method) {
+                $isFound = (bool) $this->betterNodeFinder->findFirst((array) $method->stmts, function (Node $subNode) use ($propertyName): bool {
```

Comment:
I tried using `findFirstInFunctionLikeScoped()` which check only in scoped of classmethod, but seems fails got parent type, it possibly due to classmethod of the class pulled from `AstResolver` which make the information of parent is not filled, so I used `findFirst()` instead.